src/TiciTacaToeyGameEngine.ts

```diff
@@ -147,6 +165,9 @@ class TiciTacaToeyGameEngine implements GameEngine {
           ) {
             reject({ error: ErrorCodes.INVALID_MOVE, message });
           }
+          if (this.games[message.gameId].timers[message.playerId] <= 0) {
```

Comment:
wait a minute don't you have to check the `timeLeft` property of timer to ensure player has timed out?

```typescript
if (this.games[message.gameId].timers[message.playerId].timeLeft <= 0) { // .timeLeft?
```